src/coreclr/src/vm/eventpipebuffer.cpp

```diff
@@ -23,7 +23,7 @@ EventPipeBuffer::EventPipeBuffer(unsigned int bufferSize, EventPipeThread* pWrit
     m_state = EventPipeBufferState::WRITABLE;
     m_pWriterThread = pWriterThread;
     m_eventSequenceNumber = eventSequenceNumber;
-    m_pBuffer = new BYTE[bufferSize];
+    m_pBuffer = (BYTE*)ClrVirtualAlloc(NULL, bufferSize, MEM_COMMIT, PAGE_READWRITE);
     memset(m_pBuffer, 0, bufferSize);
```

Comment:
I think we can skip the `memset` now since you used the `MEM_COMMIT` flag here:
> The function also guarantees that when the caller later initially accesses the memory, the contents will be zero.
-[docs](https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc)

Reply:
Isn't that doc for the Windows `VirtualAlloc` API? I don't think the PAL implementation memsets it to 0 by default.

Reply:
https://github.com/dotnet/runtime/blob/84ec25f8034b2072838e516849313b6c23f409cd/src/coreclr/src/pal/src/map/virtual.cpp#L1359

These comments seem to indicate it is supported, but I'll double check that we actually mimic that behavior

Reply:
Based on mmap docs and our code not passing MAP_UNINITIALIZED flag to mmap I would assume the memory we get back from VirtualAlloc is zero initialized. However I'd recommend only adding a comment to the memset referencing this discussion rather than removing it. We'd like to take this into servicing which has low tolerance for risk and we've already benchmarked the current code as being good enough as-is. In the future we might do some performance measurements to improve the throughput of EventPipe - if the memset shows up in that analysis we can address it then, if not we've saved time worrying about it.